### PR TITLE
test: expand convertToInteractiveView test suite and implement staged migration warnings

### PR DESCRIPTION
**Pull request by MasatoMakino**
## Summary

Expand convertToInteractiveView test suite from 3 to 13 comprehensive tests and implement staged migration warnings for previously unsupported re-conversion scenarios.

### 🧪 Test Suite Expansion
- **Comprehensive coverage**: Property preservation, generic types, handler initialization, edge cases
- **Console warning verification**: Test warning behavior with stderr output suppression  
- **Type safety improvements**: Replace `any` casts with proper type guards and inheritance
- **Maintainable assertions**: Use `expect.stringContaining()` for resilient warning message tests

### 🚨 Unsupported Scenario Handling
- **Detection**: Add `hasInteractionHandler` type guard to detect existing handlers
- **Staged migration**: Console warnings for re-conversion attempts with future Error behavior
- **Data preservation**: Return unchanged object instead of silent overwriting
- **Clear messaging**: English warnings with migration timeline and recommended alternatives

### 🔧 Technical Improvements
- **Type safety**: Replace `any` with `unknown` and proper type casting patterns
- **Global spy pattern**: Suppress stderr output during tests with `beforeEach` cleanup
- **Natural inheritance**: Leverage automatic upcasting from interactive mesh classes to `Mesh`
- **JSDoc updates**: Document deprecation warnings and migration guidance

## Test Coverage

```
Test Files  16 passed (16)
     Tests  141 passed (141) 
```

**convertToInteractiveView specific**: 13/13 tests passing
- ✅ Basic conversion for all converter types
- ✅ Property preservation across conversions  
- ✅ Generic value type support
- ✅ Handler initialization state verification
- ✅ Edge case handling (empty meshes, various configurations)
- ✅ Warning behavior for re-conversion scenarios
- ✅ Cross-converter warning consistency

## Migration Strategy

**Current Version**: Shows console warning, returns object unchanged
```javascript
WARNING: Cannot convert object that already has an interactionHandler. 
Returning existing object unchanged. 
This behavior will throw an Error in a future version. 
Use ClickableMesh constructor for new objects instead.
```

**Future Version**: Will throw Error (documented in warning messages)

This staged approach provides backward compatibility while guiding users toward correct usage patterns.

🤖 Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

* **New Features**
  * Added warnings when attempting to convert a mesh that is already interactive, preventing unintentional overwriting of interaction handlers.
  * Updated documentation to reflect new warning behavior during repeated conversions.

* **Bug Fixes**
  * Improved robustness by ensuring conversion functions return the original mesh unchanged if it already has an interaction handler.

* **Tests**
  * Significantly expanded test coverage for conversion functions, including property preservation, type safety, state initialization, and error handling.

### CODE REVIEW

File: __test__/convertToInteractiveView.spec.ts [hunk 1]
```diff
@@ -10,38 +10,362 @@ import {
   RadioButtonInteractionHandler,
 } from "../src/index.js";
 
+// Suppress console.warn output during tests while still allowing spy verification
+const _globalWarnSpy = vi.spyOn(console, "warn").mockImplementation(() => {});
+
```
[the remark this answers is not in the document]
Comment (MasatoMakino) [reply __test__/convertToInteractiveView.spec.ts:15]: 0b23ac5fb50fab630118741df8216c77501fcb94
で対処しました。

File: __test__/convertToInteractiveView.spec.ts [hunk 2]
```diff
@@ -10,38 +10,362 @@ import {
...
+          expect.stringContaining(
+            "WARNING: Cannot convert object that already has an interactionHandler",
+          ),
+        );
+
+        expect(
+          result.interactionHandler,
+          `${type} should return original handler unchanged after warning`,
+        ).toBe(originalHandler);
+      }
```
[the remark this answers is not in the document]
Comment (MasatoMakino) [reply __test__/convertToInteractiveView.spec.ts]: 273d1699bd6d5e35d8f6f58032795418079702af
で対処しました。
